src/v/kafka/server/usage_aggregator.cc

```diff
@@ -101,6 +101,14 @@ usage usage::operator+(const usage& other) const {
       .bytes_received = bytes_received + other.bytes_received};
 }
 
+usage& usage::operator+=(const usage& other) {
+    if (this != &other) {
+        bytes_sent += other.bytes_sent;
+        bytes_received += other.bytes_received;
+    }
```

Comment:
it seems like odd behavior to prevent adding from `this`. it'd be like an object where `1 + 1 = 1`. is there something im missing?